### ISSUE TITLE
rule: contents-no-box-props

### ISSUE THREAD

**Issue by purupurupu** [purupurupu/css-noop-checker#122]
## Rule ID
`contents-no-box-props`

## What does this rule detect?
`display: contents` eliminates the element's box entirely. All box-decoration properties (margin, padding, border, background, width, height) have no visual effect since there is no box to decorate.

## Example: should warn
```html
<div style="display: contents; margin: 10px; padding: 20px; border: 1px solid red;">
  <span>child</span>
</div>
```

## Example: should NOT warn
```html
<div style="display: flex; margin: 10px;">
  <span>child</span>
</div>
```

## Related CSS Properties
`display`, `margin`, `padding`, `border-*`, `width`, `height`, `background-*`

## References
- [CSS Display Level 3 §6.7 - display: contents](https://drafts.csswg.org/css-display/#valdef-display-contents): "element itself does not generate any boxes"
- `display: contents` is widely used for flex/grid wrapper elimination; developers often leave margin/padding not realizing the box is gone
